context-propagation/src/main/java/io/micrometer/context/ThreadLocalAccessor.java

```diff
@@ -69,4 +100,15 @@ default void restore(V previousValue) {
         setValue(previousValue);
     }
 
+    /**
+     * Remove the current {@link ThreadLocal} value when restoring values after a
+     * {@link io.micrometer.context.ContextSnapshot.Scope} closes but there was no
+     * {@link ThreadLocal} value present prior to the closed scope.
+     * @see #getValue()
+     * @since 1.0.3
+     */
+    default void restore() {
+        setValue();
+    }
```

Comment:
As per my previous comment, this would make more sense to delegate to `reset()` which would no longer be deprecated nor have a default implementation.